pkg/steps/source.go

```diff
@@ -109,38 +129,41 @@ func (s *sourceStep) Run(ctx context.Context, dry bool) error {
 		return fmt.Errorf("could not resolve clonerefs source: %v", err)
 	}
 
-	return handleBuild(s.buildClient, createBuild(s.config, s.jobSpec, clonerefsRef, s.resources), dry, s.artifactDir, s.dryLogger)
+	return handleBuild(s.buildClient, createBuild(s.config, s.jobSpec, clonerefsRef, s.resources, s.sshSecretName), dry, s.artifactDir, s.dryLogger)
 }
 
-func createBuild(config api.SourceStepConfiguration, jobSpec *api.JobSpec, clonerefsRef coreapi.ObjectReference, resources api.ResourceConfiguration) *buildapi.Build {
+func createBuild(config api.SourceStepConfiguration, jobSpec *api.JobSpec, clonerefsRef coreapi.ObjectReference, resources api.ResourceConfiguration, sshSecretName string) *buildapi.Build {
 	var refs []prowapi.Refs
 	if jobSpec.Refs != nil {
+		if len(sshSecretName) > 0 {
+			jobSpec.Refs.CloneURI = fmt.Sprintf("ssh://git@github.com/%s/%s.git", jobSpec.Refs.Org, jobSpec.Refs.Repo)
```

Comment:
Mutating the JobSpec this deep into the execution stack is very surprising. Is it necessary or could we just copy `Refs` and `ExtraRefs`? If it is, we should make the change in `main.go`.

Reply:
Yea we shouldn't mutate jobspec in this level. I addressed that.